src/main.cc

```diff
@@ -692,7 +692,10 @@ int run_client(StringView session, StringView name, StringView client_init,
     try
     {
         Optional<int> stdin_fd;
-        if (not isatty(0))
+        // json-ui is not intended to be a user interactive session
+        // so don't worry about making the tty your stdin if fd 0
+        // is not a tty.
+        if (not isatty(0) && ui_type != UIType::Json)
```

Comment:
In `create_local_ui()` Kakoune only reads stdin into a buffer for `UIType::Terminal`, rather than "everything but `UIType::Json`:

https://github.com/mawww/kakoune/blob/e7100dc87434d933fd8268e5bf70080b627750c5/src/main.cc#L644-L645

...so we should copy the same logic here too.

Reply:
To be consistent with the if condition you pointed out,

I changed:
```c++
if (not isatty(0) && ui_type != UIType::Json)
```

to:
```
if (ui_type == UIType::Terminal && not isatty(0))
```